### PR TITLE
SQL Query Changes for H2 DB Upgrade

### PR DESCRIPTION
**Pull request by ImalshaG**
This PR adds the query changes needed to upgrade the H2 DB engine from 1.4.199 to 2.1.210.

The new H2 version does not allow:
- keywords as Column names. Therefore "VALUE" column name of 5 tables are back-quoted and used, and the queries related to the new column name are changed accordingly.
- trailing commas
- data type size to be defined for BOOLEAN and INT datatypes.
- "UNIQUE KEY" keyword. Therefore it is changed to "UNIQUE" keyword.

This change is relevant only to the H2 db. Therefore a check is added where the changed sql queries are executed, in order to execute the new queries only if the db is H2.

Related to: https://github.com/wso2/product-is/issues/13162

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1821,7 +1821,7 @@
         <!-- Orbits -->
         <orbit.version.xmlschema>1.4.7.wso2v3</orbit.version.xmlschema>
         <orbit.version.infinispan>5.1.2.wso2v1</orbit.version.infinispan>
-        <orbit.version.h2.engine>1.4.199.wso2v1</orbit.version.h2.engine>
+        <orbit.version.h2.engine>2.1.210.wso2v1</orbit.version.h2.engine>
```
Comment (senthalan) [inline pom.xml:1824]: if there is no usage, let's remove this

**Review by senthalan — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java [hunk 2]
```diff
@@ -180,4 +192,13 @@
...
                     "BETWEEN ? AND ? AND IDN_AUTH_SESSION_STORE.TENANT_ID = ? ";
+
+    public static final String SQL_GET_ACTIVE_SESSION_COUNT_BY_TENANT_H2 =
+            "SELECT COUNT( DISTINCT IDN_AUTH_SESSION_META_DATA.SESSION_ID) " +
+                    "FROM IDN_AUTH_SESSION_META_DATA INNER JOIN IDN_AUTH_USER_SESSION_MAPPING " +
+                    "ON IDN_AUTH_SESSION_META_DATA.SESSION_ID = IDN_AUTH_USER_SESSION_MAPPING.SESSION_ID " +
+                    "INNER JOIN IDN_AUTH_SESSION_STORE " +
+                    "ON IDN_AUTH_USER_SESSION_MAPPING.SESSION_ID = IDN_AUTH_SESSION_STORE.SESSION_ID " +
+                    "WHERE IDN_AUTH_SESSION_META_DATA.PROPERTY_TYPE = ? AND IDN_AUTH_SESSION_META_DATA.`VALUE` " +
+                    "BETWEEN ? AND ? AND IDN_AUTH_SESSION_STORE.TENANT_ID = ? ";
```
Comment (chamathns) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java:203]: fix formatting

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/util/JdbcUtils.java [hunk 3]
```diff
@@ -17,74 +17,78 @@
...
+     * @deprecated
      */
+    @Deprecated
     public static boolean isH2MySqlOrPostgresDB() throws DataAccessException {
 
-        return isDBTypeOf(MY_SQL) || isDBTypeOf(H2) || isDBTypeOf(POSTGRE_SQL) || isDBTypeOf(MARIADB);
+        return org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB();
```
Comment (chamathns) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/util/JdbcUtils.java:52]: shall we add static imports for these? Would improve readability

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/util/JdbcUtils.java [hunk 4]
```diff
@@ -18,130 +18,143 @@
...
+     * @deprecated
      */
+    @Deprecated
     public static boolean isH2MySqlOrPostgresDB() throws DataAccessException {
 
-        return isDBTypeOf(MY_SQL) || isDBTypeOf(H2) || isDBTypeOf(POSTGRE_SQL) || isDBTypeOf(MARIA_DB);
+        return org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB();
```
Comment (chamathns) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/util/JdbcUtils.java:53]: shall we add static imports for these?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java (hunk 2 above)
Comment (ImalshaG) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java:203]: This is the format used for SQL queries.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/util/JdbcUtils.java (hunk 3 above)
Comment (ImalshaG) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/util/JdbcUtils.java:52]: Addressed by https://github.com/wso2/carbon-identity-framework/pull/3993

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/util/JdbcUtils.java (hunk 4 above)
Comment (ImalshaG) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/util/JdbcUtils.java:53]: Addressed by https://github.com/wso2/carbon-identity-framework/pull/3993.

File: components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/util/JdbcUtils.java [hunk 5]
```diff
@@ -19,62 +19,65 @@
...
+     * @deprecated
      */
+    @Deprecated
     public static boolean isH2MySqlOrPostgresDB() throws DataAccessException {
 
-        return isDBTypeOf(MY_SQL) || isDBTypeOf(H2) || isDBTypeOf(POSTGRE_SQL) || isDBTypeOf(MARIADB);
+        return org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB();
```
[the remark this answers is not in the document]
Comment (ImalshaG) [reply components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/util/JdbcUtils.java:54]: Addressed by https://github.com/wso2/carbon-identity-framework/pull/3993.

**Pull request merged**